lib/puppet/type/acl/ace.rb

```diff
@@ -29,7 +29,11 @@ def initialize(permission_hash, provider = nil)
       self.id = permission_hash['id']
       @mask = permission_hash['mask']
       self.rights = permission_hash['rights']
-      self.type = permission_hash['type']
+      if permission_hash['type'] && !permission_hash['perm_type']
+        self.perm_type = permission_hash['type']
+      else
+        self.perm_type = permission_hash['perm_type']
+      end
```

Comment:
Note the warning comments I put in. Make it clear that type will override a set perm_type.